edxval/api.py

```diff
@@ -1149,54 +1155,69 @@ def import_transcript_from_fs(edx_video_id, language_code, file_name, provider,
-        try:
-            with resource_fs.open(combine(static_dir, file_name), 'r', encoding='utf-8-sig') as f:
-                file_content = f.read()
-        except ResourceNotFound:
-            # Don't raise exception in case transcript file is not found in course OLX.
-            logger.warning(
-                '[edx-val] "%s" transcript "%s" for video "%s" is not found.',
-                language_code,
-                file_name,
-                edx_video_id
-            )
-            return
-        except UnicodeDecodeError:
-            # Don't raise exception in case transcript contains non-utf8 content.
-            logger.warning(
-                '[edx-val] "%s" transcript "%s" for video "%s" contains a non-utf8 file content.',
-                language_code,
-                file_name,
-                edx_video_id
-            )
-            return
+    # Read file from import file system and attach it to transcript record in DS.
+    try:
+        with resource_fs.open(combine(static_dir, file_name), 'r', encoding='utf-8-sig') as f:
+            file_content = f.read()
+    except ResourceNotFound:
+        # Don't raise exception in case transcript file is not found in course OLX.
+        logger.warning(
+            '[edx-val] "%s" transcript "%s" for video "%s" is not found.',
+            language_code,
+            file_name,
+            edx_video_id
+        )
+        return
+    except UnicodeDecodeError:
+        # Don't raise exception in case transcript contains non-utf8 content.
+        logger.warning(
+            '[edx-val] "%s" transcript "%s" for video "%s" contains a non-utf8 file content.',
+            language_code,
+            file_name,
+            edx_video_id
+        )
+        return
 
-        # Get file format from transcript content.
-        try:
-            file_format = get_transcript_format(file_content)
-        except Error:
-            # Don't raise exception, just don't create transcript record.
-            logger.warning(
-                '[edx-val] Error while getting transcript format for video=%s -- language_code=%s --file_name=%s',
-                edx_video_id,
-                language_code,
-                file_name
-            )
-            return
-
-        # Create transcript record.
-        create_video_transcript(
-            video_id=edx_video_id,
-            language_code=language_code,
-            file_format=file_format,
-            content=ContentFile(file_content.encode('utf-8')),
-            provider=provider
+    # change file content to utf8
+    utf8_encoded_file_content = file_content.encode('utf-8')
+    new_transcript_content_file = ContentFile(utf8_encoded_file_content)
+
+    # check if transcript content already exists
+    if (
+        existing_transcript and
+        is_duplicate_file(
+            new_transcript_content_file,
+            existing_transcript.transcript_content_hash
```

Comment:
How will this be backward compatible(given default value is '')?

Reply:
If the hash is empty, then the file would not be treated as a duplicate, and it would be uploaded.

Would you prefer that I apply the `generate_file_content_hash` on each existing uploaded file instead? The migration would take longer but it would be backwards compatible. 

> I'll add a commit for that once I target the other comments, that way if it isn't needed, we can just revert it.

Reply:
I have added backwards compatibility now :+1: sorry for not doing that earlier